lib/http/response.rb

```diff
@@ -31,12 +31,18 @@ class Response
     # @return [URI, nil]
     attr_reader :uri
 
-    def initialize(status, version, headers, body, uri = nil) # rubocop:disable ParameterLists
-      @version = version
-      @body    = body
-      @uri     = uri && HTTP::URI.parse(uri)
-      @status  = HTTP::Response::Status.new status
-      @headers = HTTP::Headers.coerce(headers || {})
+    def initialize(status, version, headers, connection_or_body, encoding = nil, uri = nil) # rubocop:disable ParameterLists
+      @version  = version
+      @uri      = uri && HTTP::URI.parse(uri)
+      @status   = HTTP::Response::Status.new status
+      @headers  = HTTP::Headers.coerce(headers || {})
+      @encoding = encoding
+
+      if connection_or_body.is_a? HTTP::Connection
```

Comment:
To the above, moving this to a hash means just setting it based on key rather than having to do content type detection.